third_party/amd/lib/TritonAMDGPUTransforms/ConvertToBufferOps.cpp

```diff
@@ -38,10 +38,259 @@ namespace ttg = mlir::triton::gpu;
 namespace tt = mlir::triton;
 
 namespace {
-template <typename F>
-bool verifyNonSmallerByAssumption(Value expr,
-                                  const DenseSet<Value> &assumptions,
-                                  F matchesOther) {
+
+constexpr int64_t kDefaultMaxTripCount = 1024;
```

Comment:
This is fine for now. But before turning it on as default, we need to measure the impact on compilation time and adjust I think.